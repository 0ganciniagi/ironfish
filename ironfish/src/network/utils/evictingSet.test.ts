/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/. */

import EvictingSet from './evictingSet'

it('Constructs an EvictingSet with no elements', () => {
  const set = new EvictingSet(5)
  expect(set).toMatchSnapshot()
})

it('Adds an element to an EvictingSet', () => {
  const set = new EvictingSet(5)
  set.add('a')
  expect(set).toMatchSnapshot()
})

it('Removes an element once max size is reached', () => {
  const set = new EvictingSet(5)
  set.add('a')
  set.add('b')
  set.add('c')
  set.add('d')
  set.add('e')
  set.add('f')
  set.add('g')
  set.add('h')
  set.add('i')
  set.add('j')
  set.add('k')
  expect(set).toMatchSnapshot()
})
