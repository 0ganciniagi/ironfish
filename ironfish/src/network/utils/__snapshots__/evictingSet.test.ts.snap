// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adds an element to an EvictingSet 1`] = `
EvictingSet {
  "items": Set {
    "a",
  },
  "max_size": 5,
}
`;

exports[`Constructs an EvictingSet with no elements 1`] = `
EvictingSet {
  "items": Set {},
  "max_size": 5,
}
`;

exports[`Removes an element once max size is reached 1`] = `
EvictingSet {
  "items": Set {
    "g",
    "h",
    "i",
    "j",
    "k",
  },
  "max_size": 5,
}
`;
